src/stakers/stakers.controller.ts

```diff
@@ -22,6 +23,46 @@ import { StakersService } from './stakers.service';
 export class StakersController {
   constructor(private readonly stakersService: StakersService) {}
 
+
+  @Get('/ns/all')
+  @ApiOperation({
+    summary: 'Fetch Native Stake balances for a specific date interval',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'The records were successfully fetched.',
+    type: NativeStakeBalanceDto,
```

Comment:
Using `NativeStakeBalanceDto` and `NativeStakeBalancesDto` can be dangerous exactly for these reasons, this line is supposed to contain `NativeStakeBalancesDto` (the newly added DTO). This line should definitely be fixed, but I'd also recommend renaming the new DTO to something like `NativeStakeAllBalancesDto` or `AllNativeStakeBalancesDto` to differentiate it a bit more.